packages/common/src/llm/schemas.ts

```diff
@@ -0,0 +1,99 @@
+import { z } from '@botpress/sdk'
+
+const ToolCallSchema = z.object({
+  id: z.string(),
+  type: z.enum(['function']),
+  function: z.object({
+    name: z.string(),
+    arguments: z.string(),
+  }),
+})
+
+export type ToolCall = z.infer<typeof ToolCallSchema>
+
+const ToolChoiceSchema = z.object({
+  // TODO: remove empty value from enum once Studio issue is fixed
+  type: z.enum(['auto', 'specific', 'any', 'none', '']).optional(), // note: Claude doesn't support "none" but we can simply strip out the tools when `type` is "none"
+  functionName: z.string().optional().describe('Required if `type` is "specific"'),
+})
+
+const MessageSchema = z.object({
+  role: z.enum(['user', 'assistant']),
+  type: z.enum(['text', 'tool_calls', 'tool_result', 'multipart']),
+  toolCalls: z.array(ToolCallSchema).optional().describe('Required if `type` is "tool_calls"'),
+  toolResultCallId: z.string().optional().describe('Required if `type` is "tool_result"'), // note: not supported by Gemini
+  content: z
+    .string()
+    // TODO: We need to check limits of integration action input/output in the bridge
+    // TODO: union types are not supported yet by the Studio, this needs to be fixed before we enable multipart support
+    // .or(
+    //   z.array(
+    //     z.object({
+    //       type: z.enum(['text', 'image']),
+    //       mimeType: z.string(),
+    //       data: z.string().describe('plain text or base64-encoded binary data'),
```

Comment:
**issue:** I talked about this with mike and we both believe it's a bad idea to let the image bytes pass by the bridge. The best approach would be to stream these bytes to file API (with `client.uploadFile`) and return an imageUrl

Reply:
This will leave out all the LLMs out there that don't support URIs for media (one big one is Claude). If we truly want to become multi-LLM, we should plan to address this limitation/concern on the bridge.

Reply:
Actually, I think we can indeed just pass the URI (whether external or from the FIles API) and let the integration retrieve the file and pass the raw data to the LLM... I'll refactor this type to have `content` for "text" parts and `uri` for "image" parts.

Reply:
That's not what I meant: I didn't mean the model should not return bytes, I meant the integration should be responsible for uploading it to the files API before sending back the response